### PR TITLE
Calendly Extension

### PR DESCRIPTION
**Pull request by eluce2**
## Description

This calendly extension will allow Calendly users to quickly get the schedulign URL of any of their event types configured in Calendly. A secondary action allows users to general a single-use link for any event. In both cases the result is having the scheduling URL copied to the clipboard.

## Type of change

*New extension*

## Screencast

[Video Link](https://www.dropbox.com/s/8mge50cfmhtaavk/CleanShot%202021-10-15%20at%2020.37.25.mp4?dl=0)

## Checklist

- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/calendly/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "title": "Calendly",
+  "description": "Raycast commands to access Calendly instance",
+  "icon": "logo.png",
+  "author": "eluce2",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "calendly",
+      "title": "Calendly",
+      "subtitle": "Copy Meeting URL",
```
Comment (thomaspaulmann) [inline extensions/calendly/package.json:12]: We do the command titles the other way around ;-) Usually we have some `<Verb> <Noun>` as command title and use the service as subtitle. How about something like this?

```suggestion
      "title": "Share Meeting Link",
      "subtitle": "Calendly",
```

File: extensions/calendly/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "calendly",
+  "title": "Calendly",
```
Comment (thomaspaulmann) [inline extensions/calendly/package.json:3]: Should we name this a little bit more precise? It's the title of the store entry.

```suggestion
  "title": "Calendly Share Meeting Links",
```

File: extensions/calendly/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  "description": "Raycast commands to access Calendly instance",
```
Comment (thomaspaulmann) [inline extensions/calendly/package.json:4]:
```suggestion
  "description": "Quickly share your Calendly meeting links",
```

File: extensions/calendly/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "@raycast/api": "^1.25.0",
+    "axios": "^0.23.0",
+    "lodash": "^4.17.21"
+  },
+  "preferences": [
+    {
+      "name": "token",
+      "type": "password",
+      "required": true,
+      "title": "Calendly Personal Access Token",
```
Comment (thomaspaulmann) [inline extensions/calendly/package.json:27]: Given that this is already in the Calendly context, I would remove the suffix. This should also make it fit on one line.
```suggestion
      "title": "Personal Access Token",
```

**Comment by eluce2** [comment]
@thomaspaulmann Ok I've added some caching behavior, but I'm certainly still open to more ideas if you have another suggestion for the implementation.

**Comment by thomaspaulmann** [comment]
> Yes, I can add caching, but I didn't see an option for easily expiring the cache. Is there a way to easily expose a way to update a stale cache, even if it's only daily or weekly? I didn't see this in the figma example

Figma uses SWR behavior. So it loads the old cache and then refreshes with the actual data from the network and updates the cache. This way, the cache is less often outdated if you use the commands often. It can be improved to ignore the cache after some expiration time.

**Pull request merged**